# Chi-Square test


# Data Provided:

The data is summarized in a contingency table showing the counts of customers in each satisfaction level for both types of devices:

Satisfaction	Smart Thermostat	Smart Light	Total

Very Satisfied	50	70	120

Satisfied	80	100	180

Neutral	60	90	150

Unsatisfied	30	50	80

Very Unsatisfied	20	50	70

Total	240	360	600

# Objective:

To use the Chi-Square test for independence to determine if there's a significant association between the type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level.


# 1. State the Hypotheses:


In [3]:
# Define the hypotheses as strings
null_hypothesis = "H0: There is no association between the type of smart home device and customer satisfaction level."
alternative_hypothesis = "Ha: There is an association between the type of smart home device and customer satisfaction level."

# Print the hypotheses
print(null_hypothesis)
print(alternative_hypothesis)

H0: There is no association between the type of smart home device and customer satisfaction level.
Ha: There is an association between the type of smart home device and customer satisfaction level.


# 2. Compute the Chi-Square Statistic:


In [5]:
import numpy as np

# Observed frequencies from the contingency table
observed_frequencies = np.array([
    [50, 70],
    [80, 100],
    [60, 90],
    [30, 50],
    [20, 50]
])

# Calculate expected frequencies
row_totals = observed_frequencies.sum(axis=1)
col_totals = observed_frequencies.sum(axis=0)
total = observed_frequencies.sum()

expected_frequencies = np.outer(row_totals, col_totals) / total

# Compute Chi-Square statistic for each cell
chi_square_values = ((observed_frequencies - expected_frequencies) ** 2) / expected_frequencies

# Sum up all Chi-Square values to get the total Chi-Square statistic
chi_square_statistic = chi_square_values.sum()

print("Chi-Square Statistic:", chi_square_statistic)

Chi-Square Statistic: 5.638227513227513


# 3. Determine the Critical Value:
Using the significance level (alpha) of 0.05 and the degrees of freedom (which is the number of categories minus 1)


In [6]:
from scipy.stats import chi2

# Significance level (alpha)
alpha = 0.05

# Degrees of freedom (number of categories minus 1)
degrees_of_freedom = (5 - 1) * (2 - 1)

# Calculate the critical value
critical_value = chi2.ppf(1 - alpha, degrees_of_freedom)

print("Critical Value:", critical_value)

Critical Value: 9.487729036781154


Since 
5.64 is less than 
9.488, we fail to reject the null hypothesis.

# 4. Make a Decision:


# Decision:

We do not have enough evidence to reject the null hypothesis. Therefore, we conclude that there is no significant association between the type of smart home device and customer satisfaction levels at the 
0.05 significance level.

This means that we cannot conclude that there is a difference in satisfaction levels between Smart Thermostats and Smart Lights based on the data provided.